.github/workflows/build_and_release.yml

```diff
@@ -0,0 +1,42 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract version from file
```

Comment:
Extract this from the git `main` branch tag?
Cilium's workflows does this:
https://github.com/cilium/cilium/blob/896d0420f11c8eefe317e77a2a43805e63f2c3bd/.github/workflows/build-images-ci.yaml#L103-L115
https://github.com/cilium/cilium/blob/896d0420f11c8eefe317e77a2a43805e63f2c3bd/.github/workflows/build-images-ci.yaml#L171-L173

BTW they also do image signing using `cosign`, which you might want to look into:
https://github.com/cilium/cilium/blob/896d0420f11c8eefe317e77a2a43805e63f2c3bd/.github/workflows/build-images-ci.yaml#L221-L230